app/controllers/api/v1/admin/organizations_controller.rb

```diff
@@ -0,0 +1,41 @@
+module Api
+  module V1
+    module Admin
+      class OrganizationsController < ApiController
+        before_action :authenticate!
+        before_action :authorize_super_admin
+
+        def update
+          organization = Organization.find(params[:id])
+          organization.assign_attributes(organization_params)
+          if organization.save
+            render json: {
+              id: organization.id,
+              name: organization.name,
+              profile_image: organization.profile_image,
+              slug: organization.slug,
+              summary: organization.summary,
+              tag_line: organization.tag_line,
+              url: organization.url
+            }, status: :ok
+          else
+            render json: { error: organization.errors_as_sentence, status: 422 }, status: :unprocessable_entity
+          end
+        end
+
+        def destroy
+          organization = Organization.find(params[:id])
+          organization.destroy
```

Comment:
(question, maybe blocking?) Do we not need to implement more than just a delete here? Currently when we delete organizations, we have a delete workers that disassociates the organization from the articles, notifications etc. 

- Delete worker can be found [here](https://github.com/forem/forem/blob/dukegreene/organization-api-endpoints/app/workers/organizations/delete_worker.rb#L27).
-  Deleter service can be found [here](https://github.com/forem/forem/blob/dukegreene/organization-api-endpoints/app/services/organizations/delete.rb#L1)

Reply:
Thanks! I'll investigate this and see what's necessary. 

I had also meant to ask you, based on the work you've done digging into how associated credits can stop an org from being deleted... should this route have a more specific error come back in the event an org fails to delete because it still has unspent credits associated with it?

Reply:
>  should this route have a more specific error come back in the event an org fails to delete because it still has unspent credits associated with it?

I think so, you could probably rescue the error and re-surface it in some meaningful way. So for example if I have an organization with credits and do the following: 

```
begin
  org_with_credits = Organization.find(5)
  org_with_credits.destroy!
rescue ActiveRecord::RecordNotDestroyed => error
``` 

The error I get back is one that tells me I have a `restrict_with_error`  on credits. 
```
org.errors
=> #<ActiveModel::Errors [#<ActiveModel::Error attribute=base, type=restrict_dependent_destroy.has_many, options={:record=>"credits"}>]>
```

Reply:
Oooh and sorry one more detail that's not directly related but may help avoid confusion - the `restrict_with_error` doesn't only fail if there are _unspent credits_ but just if there are credits that were associated with the record in general even if they are spent already. SInce the plan is to remove credits in a stable way at some point I didn't dive deeper into evaluating/changing that logic but just kept it as is.